test/graphql/plug/endpoint_test.exs

```diff
@@ -46,10 +47,23 @@ defmodule GraphQL.Plug.EndpointTest do
     assert_query {:post, "/", query: "{greeting}"}, {200, success}
   end
 
+  test "root data can be set at request time" do
+    defmodule TestRootPlug do
+      use Plug.Builder
+      def root_eval(_conn), do: %{greeting: "Root"}
+      plug GraphQL.Plug.Endpoint, [schema: {TestSchema, :schema}, root: &TestRootPlug.root_eval/1 ]
```

Comment:
A note regarding this, currently the function is being pulled out and executed in the `init` function for grabbing the schema. For this function, I will need to make the evaluation and execution happen in `call` instead. Otherwise, we would never execute it per request or get the `conn` reference.